networkx/algorithms/polynomials.py

```diff
@@ -0,0 +1,145 @@
+"""Provides algorithms supporting the computation of graph polynomials.
+
+Graph polynomials are polynomial-valued graph invariants that encode a wide
+variety of structural information. Examples include the Tutte polynomial,
+chromatic polynomial, characteristic polynomial, and matching polynomial. An
+extensive treatment is provided in [1]_.
+
+.. [1] Y. Shi, M. Dehmer, X. Li, I. Gutman,
+   "Graph Polynomials"
+"""
+from collections import deque
+import networkx as nx
+from networkx.utils import not_implemented_for
+
+__all__ = ["tutte_polynomial"]
+
+
+@not_implemented_for("directed")
+def tutte_polynomial(G):
+    r"""Compute a graph's Tutte polynomial via an iterative version of the
+    deletion-contraction algorithm.
+
+    The Tutte polynomial `T_G(x, y)` is a fundamental graph polynomial invariant in
+    two variables. It encodes a wide array of information related to the
+    edge-connectivity of a graph; "Many problems about graphs can be reduced to
+    problems of finding and evaluating the Tutte polynomial at certain values" [1]_.
+    In fact, every deletion-contraction-expressible feature of a graph is a
+    specialization of the Tutte polynomial [2]_ (see Notes for examples).
+
+    There are several equivalent definitions; here are three:
+
+    Def 1 (rank-nullity expansion): For `G` an undirected graph, `n(G)` the
+    number of vertices of `G`, `E` the edge set of `G`, and `c(A)`` the number of
+    connected components of `A` [3]_:
```

Comment:
In the formula, `A` is a subset of edges, but `c(A)` is the number of connected components of `A`. What is the graph used in `c(A)`?

Reply:
Good point. `c(A)` counts the connected components of the graph with the edge set `A` and the same vertex set as `G`. I will clarify this.